src/cz/crcs/ectester/standalone/libs/jni/NativeKeyAgreementSpi.java

```diff
@@ -373,4 +374,56 @@ public MatrixsslECDH() {
         }
     }
 
+    public abstract static class Nettle extends SimpleKeyAgreementSpi {
+        private String type;
+
+        public Nettle(String type) {
+            this.type = type;
+        }
+
+        @Override
+        byte[] generateSecret(byte[] pubkey, byte[] privkey, ECParameterSpec params) {
+            try {
+                AlgorithmParameters tmp = AlgorithmParameters.getInstance("EC");
+                tmp.init(params);
+                ECGenParameterSpec spec = tmp.getParameterSpec(ECGenParameterSpec.class);
+                switch (spec.getName()) {
```

Comment:
Same as the lookup in ECDSA, just noting for myself to add the reverse `EC_Store` lookup.